confidential-data-hub/storage/src/volume_type/rbd/mod.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2024 Intel
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+pub mod error;
+
+use async_trait::async_trait;
+use error::{RbdError, Result};
+use log::error;
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use tokio::{
+    fs,
+    io::{AsyncReadExt, AsyncWriteExt},
+    process::Command,
+};
+
+use super::SecureMount;
+
+/// Encrypt storage binary
+const ENCRYPT_STORAGE_BIN: &str = "/usr/local/bin/encrypt-storage";
+
+#[derive(Serialize, Deserialize, PartialEq, Debug)]
+struct RbdParameters {
+    /// The device number, formatted as "MAJ:MIN".
+    #[serde(rename = "deviceId")]
+    pub device_id: String,
+
+    /// The encryption type. Currently, only LUKSfs is supported.
+    #[serde(rename = "encryptType")]
+    pub encrypt_type: String,
+
+    /// Indicates whether the encryption key is generated by CDH or set by users.
+    #[serde(rename = "randomEncrypt")]
+    pub random_encrypt: String,
+
+    /// Encryption key if `random_encrypt` is "false".
+    #[serde(rename = "encryptKey")]
+    pub encrypt_key: Option<String>,
+
+    /// Indicates whether to enable dm-integrity.
+    #[serde(rename = "dataIntegrity")]
+    pub data_integrity: String,
+}
+
+pub(crate) struct Rbd;
+
+async fn random_encrypt_key() -> anyhow::Result<String> {
+    let storage_key_path = "/tmp/random_encrypt_rbd.key";
```

Comment:
Do we support multiple multiple encrypted rbd mounts? If so, do we need different keys?